File: MyClub/MyClub.Model/Requests/DashboardReportRequest.cs
using System;

namespace MyClub.Model.Requests
{
    /// <summary>
    /// Type of dashboard report to generate.
    /// </summary>
    public enum DashboardReportType
    {
        Top10MostSoldProducts = 1,
        Top10LeastSoldProducts = 2,
        MembershipsPerMonth = 3,
        RevenuePerMonth = 4
    }

    /// <summary>
    /// Request model used when generating a dashboard PDF.
    /// For now it only contains the report type but can be extended
    /// with date ranges, filters, localization, etc.
    /// </summary>
    public class DashboardReportRequest
    {
        public DashboardReportType Type { get; set; }
    }
}

File: MyClub/UI/myclub_desktop/lib/models/dashboard/dashboard_report_request.dart
// Model and helpers for Dashboard PDF generation requests

enum DashboardReportType {
  top10MostSoldProducts, // 1
  top10LeastSoldProducts, // 2
  membershipsPerMonth, // 3
  revenuePerMonth, // 4
}

extension DashboardReportTypeX on DashboardReportType {
  int get value {
    switch (this) {
      case DashboardReportType.top10MostSoldProducts:
        return 1;
      case DashboardReportType.top10LeastSoldProducts:
        return 2;
      case DashboardReportType.membershipsPerMonth:
        return 3;
      case DashboardReportType.revenuePerMonth:
        return 4;
    }
  }

  String get label {
    switch (this) {
      case DashboardReportType.top10MostSoldProducts:
        return 'Top 10 najprodavanijih';
      case DashboardReportType.top10LeastSoldProducts:
        return 'Top 10 najmanje prodavanih';
      case DashboardReportType.membershipsPerMonth:
        return 'Članstva po mjesecima';
      case DashboardReportType.revenuePerMonth:
        return 'Zarada po mjesecima';
    }
  }
}

class DashboardReportRequest {
  final DashboardReportType type;
  DashboardReportRequest(this.type);

  Map<String, dynamic> toJson() => {
        'type': type.value,
      };
}
